R/colourInput.R

```diff
@@ -221,10 +221,16 @@ formatHEXsingle <- function(x) {
   if (x %in% grDevices::colors()) {
     x <- do.call(grDevices::rgb, as.list(grDevices::col2rgb(x) / 255))
   }
-  if (!grepl("^#?([[:xdigit:]]{3}|[[:xdigit:]]{6})$", x)) {
+  if (!grepl("^#?([[:xdigit:]]{3}|[[:xdigit:]]{6}|[[:xdigit:]]{8})$", x)) {
     stop(sprintf("%s is not a valid colour", x), call. = FALSE)
   }
 
+  # check whether it is 6-digit hex code with alpha.
+  if (substr(x, 1, 1) == "#" && nchar(x) == 9) {
```

Comment:
correct. You can actually type "abc" into the text input itself and it will automatically correct it to "#AABBCC"